core/controllers/classifier.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for communicating with the VM for training classifiers."""
+
+import hashlib
+import hmac
+import json
+
+from core.controllers import base
+from core.domain import classifier_domain
+from core.domain import classifier_services
+from core.domain import config_domain
+
+import feconf
+
+
+def generate_signature(secret, message):
+    """Generates digital signature for given data.
+    This function should be in sync with its counterpart in Oppia-ml.
+
+    Args:
+        secret: str. The secret used to communicate with Oppia-ml.
+        message: dict. The message payload data.
+
+    Returns:
+        str. The signature of the payload data.
+    """
+    message_json = json.dumps(message, sort_keys=True)
+    return hmac.new(secret, message_json, digestmod=hashlib.sha256).hexdigest()
+
+
+def validate_message(message):
```

Comment:
validate_job_result_message_dict

(since this is only specific to job-related messages, right?)